homeassistant/components/deconz/sensor.py

```diff
@@ -124,6 +126,39 @@ class DeconzSensorDescription(SensorEntityDescription, DeconzSensorDescriptionMi
         state_class=SensorStateClass.MEASUREMENT,
         native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
     ),
+    DeconzSensorDescription[AirQuality](
+        key="air_quality_formaldehyd",
+        supported_fn=lambda device: device.air_quality_formaldehyde is not None,
```

Comment:
If you do not use the `supports_air_quality_formaldehyde` and similar in the other descriptions those could be removed from the implementation all together?

Reply:
TLDR: Yes :)

Well, I added them as requested by you (probably a misunderstanding).
I think it's a case-by-case decision based on the sensor to support.
In terms of "air_quality" more logic is required, therefore the "supports_air_quality" totally makes sense (lamda doesnt work since it always returns a value, even if there is no data - at least "unkown").
In case of "air_quality_formaldehyde" (and the other sensors) it's not required. If there is no data "None" will be returned.
To avoid more misunderstandings, with your "ok" I'll remove the following functions:

<ins>previously already present, but not used:</ins>
- supports_air_quality_ppb
- supports_pm_2_5

<ins>new in this PR (but then never merged):</ins>
- supports_air_quality_co2
- supports_air_quality_formaldehyde

Reply:
Ok! :)

Reply:
done, above functions have been removed